pkg/R/deserialize.R

```diff
@@ -122,21 +123,24 @@ readRowList <- function(obj) {
   # the numCols bytes inside the read function in order to correctly
   # deserialize the row.
   rawObj <- rawConnection(obj, "r+")
-  numCols <- SparkR:::readInt(rawObj)
-  rowOut <- SparkR:::readRow(rawObj, numCols)
-  close(rawObj)
-  rowOut
+  on.exit(close(rawObj))
```

Comment:
It doesn't reuse it right now. I was wondering if we could reuse it by passing it as an argument. Using `on.exit` as a replacement for `finally` is fine.